### DISCUSSION TITLE
VHUB not visible under Network Blocks

### DISCUSSION THREAD

**Discussion by tpetersvw** [Azure/ipam#203] [General]
I can see my VHUB with its adress space, but it isnt assigned with my ip block (htttps://ipam/discover/vhub : Block: <Unassigned>, causing overlap issues. Whats the recommended way to assign / exclude it ?

Edit:

- We have a multi tenant deployment of the solution (4 times) and in only one of them the vhub isnt visible in the "Block Networks" associaton menu under Configure. It is visible under vhubs though, any thoughts ?
- In the other 3 the vhub is visible under Network Blocks and i have assigned them succesfully now.
- a start/stop of the webapp didn't change the visibility of the vhub under Network Blocks

Solution:

- added it via the API, this succeeded.

POST 

/api/spaces/{space}/blocks/{block}/networks

body:

{
  "id": "string", << azure resource id of vhub
  "active": true
}

**Comment by DCMattyG** [comment]
Hey there @tpetersvw, this is certainly not the expected behavior.

Are you able to share some more details around the CIDR of your **Block** and the CIDR of the **vWAN Hub** that wouldn't appear on the Networking Associations page?

You should see a list of Virtual Networks and Virtual Hubs that meet these criteria:
1. Address Space of the Virtual Network or Virtual Hub falls within the CIDR of the **Block**
2. Do not overlap an **External Network** defined on the **Block**
3. Do not overlap an un-fulfilled **Reservation** on the **Block**

Additionally, in your setup are you using multiple **Spaces**, or a single **Space**?

Thanks you for the information and for reaching out. If there is an issue, lets make sure it gets fixed for everyone!

**Reply by tpetersvw** [reply]
Block Name: $tenantname-prd

Parent Space: $tenantname-prd

CIDR: 10.$tenantcidr.0.0/16

so in summary:

one space, with one /16 block, there were and are no external network definitions.

the vhub:

$tenantname-p-vhub-we
private address space: 10.$tenantcidr.0.0/23

as i said, it added correctly through api ,but wasnt visible in the webui. The last bit of info that might be relevant is we have an automated solution to deploy workloads/landingzones which , using ipam created a network with an overlapping space (10.$tenantcidr.0.8/29) that resulted in an error that pointed us to this problem (the vhub not being associated with the space/block). After the association of the vhub ipam now give out the correct cidr blocks without overlap.
